src/sets.jl

```diff
@@ -4,25 +4,51 @@
 # Use of this source code is governed by an MIT-style license that can be found
 # in the LICENSE.md file or at https://opensource.org/licenses/MIT.
 
-# Sets
-
 # Note: When adding a new set, also add it to Utilities.Model.
 
 """
     AbstractSet
 
-Abstract supertype for set objects used to encode constraints. A set object
-should not contain any [`VariableIndex`](@ref) or [`ConstraintIndex`](@ref)
-as the set is passed unmodifed during [`copy_to`](@ref).
+Abstract supertype for set objects used to encode constraints.
+
+!!! note
+    When developing new sets, the set struct must not contain any
+    [`VariableIndex`](@ref) or [`ConstraintIndex`](@ref) objects, because the
+    set is passed unmodifed during [`copy_to`](@ref).
+
+## Required methods
+
+For sets of type `S` with `isbitstype(S) == false`, you must implement:
+
+ * `Base.copy(set::S)`
+ * `Base.:(==)(x::S, y::S)`
+
+Subtypes of `AbstractSet` such as [`AbstractScalarSet`](@ref) and
+[`AbstractVectorSet`](@ref) may prescribe additional required methods.
+
+## Optional methods
+
+You may optionally implement:
+
+ * [`dual_set`](@ref)
```

Comment:
I would add `set_dot`

Reply:
Saying to only implement it in case the dot product is not `LinearAlgebra.dot` between the vectors